@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -502,20 +495,31 @@ function placePointLabelChoosingAnchor(
 
         // Glyphs arrangement have been changed remove text buffer object and apply
         // new text layout to the label.
-        if (i !== 0) {
+        if (!recentPlacementTest) {
             label.textBufferObject = undefined;
         }
+
+        if (!basePlacementTest) {
+            // Text elements with alternative placement require their own unique layout style.
+            // Provide private copy of label layout and attach it to current label's text canvas.
+            textCanvas.textLayoutStyle = label.layoutStyle!.clone();
```

Comment:
We should only need to clone the layout style at most once per label. Instead, this code here is cloning the layout style  every time a label is not placed in the base placement. This must be hurting performance really badly, especially in the current code, where this clone is done always at the beginning of this function.

TextCanvas should have it's own textRenderStyle and textLayoutStyle objects. Whenever a label needs to be processed by text canvas, we copy the label's style by doing:
textCanvas.textLayoutStyle.copy(label.layoutStyle)

where we copy every parameter, without allocating new objects.

Reply:
Done.

Reply:
Fixed. TextCanvas holds its own copy of layout / render styles.